pkg/devfile/adapters/common/command.go

```diff
@@ -164,6 +179,35 @@ func validateCommand(data data.DevfileData, command common.DevfileCommand) (err
 	return
 }
 
+// validateCompositeCommand checks that the specified composite command is valid
+func validateCompositeCommand(data data.DevfileData, compositeCommand *common.Composite) error {
+	if compositeCommand.Group.Kind == common.RunCommandGroupType {
+		return fmt.Errorf("compoosite commands cannot belong to the run grouop")
+	}
+
+	// Loop over the commands and validate that each command points to a command that's in the devfile
+	for _, command := range compositeCommand.Commands {
+		if command == compositeCommand.Id {
+			return fmt.Errorf("the composite command %q cannot reference itself", compositeCommand.Id)
+		}
+
+		if !hasCommand(data, command) {
```

Comment:
Maybe store in a map and search for faster searching

Reply:
Ah, good idea! I did that for the other instances where I had to search for a command, I should do the same here.